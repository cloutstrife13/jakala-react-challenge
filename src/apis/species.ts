import axios from "axios";
import { SwapiEndpoint } from "../utils/enum";

export const fetchStarWarsSpecies = async (speciesId?: number) => {
  if (speciesId) {
    const result = await axios.get(`${SwapiEndpoint.SPECIES}/${speciesId}`);
    return result.data;
  }

  const result = await axios.get(`${SwapiEndpoint.SPECIES}`);
  return result.data;
};
