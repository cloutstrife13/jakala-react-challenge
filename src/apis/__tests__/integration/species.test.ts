import { describe, expect, it } from "vitest";
import { fetchStarWarsSpecies } from "../../species";

describe("Species API", () => {
  it("fetches species data from SWAPI", async () => {
    const result = await fetchStarWarsSpecies();

    expect(result).toMatchSnapshot();
  });

  it("fetches a species entry from SWAPI", async () => {
    const result = await fetchStarWarsSpecies(3);

    expect(result).toMatchSnapshot();
  });
});
