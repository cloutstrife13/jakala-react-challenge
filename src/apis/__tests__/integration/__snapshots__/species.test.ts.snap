// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Species API > fetches a species entry from SWAPI 1`] = `
{
  "average_height": "210",
  "average_lifespan": "400",
  "classification": "mammal",
  "created": "2014-12-10T16:44:31.486000Z",
  "designation": "sentient",
  "edited": "2014-12-20T21:36:42.142000Z",
  "eye_colors": "blue, green, yellow, brown, golden, red",
  "films": [
    "https://swapi.dev/api/films/1/",
    "https://swapi.dev/api/films/2/",
    "https://swapi.dev/api/films/3/",
    "https://swapi.dev/api/films/6/",
  ],
  "hair_colors": "black, brown",
  "homeworld": "https://swapi.dev/api/planets/14/",
  "language": "Shyriiwook",
  "name": "Wookie",
  "people": [
    "https://swapi.dev/api/people/13/",
    "https://swapi.dev/api/people/80/",
  ],
  "skin_colors": "gray",
  "url": "https://swapi.dev/api/species/3/",
}
`;

exports[`Species API > fetches species data from SWAPI 1`] = `
{
  "count": 37,
  "next": "https://swapi.dev/api/species/?page=2",
  "previous": null,
  "results": [
    {
      "average_height": "180",
      "average_lifespan": "120",
      "classification": "mammal",
      "created": "2014-12-10T13:52:11.567000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.136000Z",
      "eye_colors": "brown, blue, green, hazel, grey, amber",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/4/",
        "https://swapi.dev/api/films/5/",
        "https://swapi.dev/api/films/6/",
      ],
      "hair_colors": "blonde, brown, black, red",
      "homeworld": "https://swapi.dev/api/planets/9/",
      "language": "Galactic Basic",
      "name": "Human",
      "people": [
        "https://swapi.dev/api/people/66/",
        "https://swapi.dev/api/people/67/",
        "https://swapi.dev/api/people/68/",
        "https://swapi.dev/api/people/74/",
      ],
      "skin_colors": "caucasian, black, asian, hispanic",
      "url": "https://swapi.dev/api/species/1/",
    },
    {
      "average_height": "n/a",
      "average_lifespan": "indefinite",
      "classification": "artificial",
      "created": "2014-12-10T15:16:16.259000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.139000Z",
      "eye_colors": "n/a",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/4/",
        "https://swapi.dev/api/films/5/",
        "https://swapi.dev/api/films/6/",
      ],
      "hair_colors": "n/a",
      "homeworld": null,
      "language": "n/a",
      "name": "Droid",
      "people": [
        "https://swapi.dev/api/people/2/",
        "https://swapi.dev/api/people/3/",
        "https://swapi.dev/api/people/8/",
        "https://swapi.dev/api/people/23/",
      ],
      "skin_colors": "n/a",
      "url": "https://swapi.dev/api/species/2/",
    },
    {
      "average_height": "210",
      "average_lifespan": "400",
      "classification": "mammal",
      "created": "2014-12-10T16:44:31.486000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.142000Z",
      "eye_colors": "blue, green, yellow, brown, golden, red",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/6/",
      ],
      "hair_colors": "black, brown",
      "homeworld": "https://swapi.dev/api/planets/14/",
      "language": "Shyriiwook",
      "name": "Wookie",
      "people": [
        "https://swapi.dev/api/people/13/",
        "https://swapi.dev/api/people/80/",
      ],
      "skin_colors": "gray",
      "url": "https://swapi.dev/api/species/3/",
    },
    {
      "average_height": "170",
      "average_lifespan": "unknown",
      "classification": "sentient",
      "created": "2014-12-10T17:05:26.471000Z",
      "designation": "reptilian",
      "edited": "2014-12-20T21:36:42.144000Z",
      "eye_colors": "black",
      "films": [
        "https://swapi.dev/api/films/1/",
      ],
      "hair_colors": "n/a",
      "homeworld": "https://swapi.dev/api/planets/23/",
      "language": "Galatic Basic",
      "name": "Rodian",
      "people": [
        "https://swapi.dev/api/people/15/",
      ],
      "skin_colors": "green, blue",
      "url": "https://swapi.dev/api/species/4/",
    },
    {
      "average_height": "300",
      "average_lifespan": "1000",
      "classification": "gastropod",
      "created": "2014-12-10T17:12:50.410000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.146000Z",
      "eye_colors": "yellow, red",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/3/",
      ],
      "hair_colors": "n/a",
      "homeworld": "https://swapi.dev/api/planets/24/",
      "language": "Huttese",
      "name": "Hutt",
      "people": [
        "https://swapi.dev/api/people/16/",
      ],
      "skin_colors": "green, brown, tan",
      "url": "https://swapi.dev/api/species/5/",
    },
    {
      "average_height": "66",
      "average_lifespan": "900",
      "classification": "mammal",
      "created": "2014-12-15T12:27:22.877000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.148000Z",
      "eye_colors": "brown, green, yellow",
      "films": [
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/4/",
        "https://swapi.dev/api/films/5/",
        "https://swapi.dev/api/films/6/",
      ],
      "hair_colors": "brown, white",
      "homeworld": "https://swapi.dev/api/planets/28/",
      "language": "Galactic basic",
      "name": "Yoda's species",
      "people": [
        "https://swapi.dev/api/people/20/",
      ],
      "skin_colors": "green, yellow",
      "url": "https://swapi.dev/api/species/6/",
    },
    {
      "average_height": "200",
      "average_lifespan": "unknown",
      "classification": "reptile",
      "created": "2014-12-15T13:07:47.704000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.151000Z",
      "eye_colors": "yellow, orange",
      "films": [
        "https://swapi.dev/api/films/2/",
      ],
      "hair_colors": "none",
      "homeworld": "https://swapi.dev/api/planets/29/",
      "language": "Dosh",
      "name": "Trandoshan",
      "people": [
        "https://swapi.dev/api/people/24/",
      ],
      "skin_colors": "brown, green",
      "url": "https://swapi.dev/api/species/7/",
    },
    {
      "average_height": "160",
      "average_lifespan": "unknown",
      "classification": "amphibian",
      "created": "2014-12-18T11:09:52.263000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.153000Z",
      "eye_colors": "yellow",
      "films": [
        "https://swapi.dev/api/films/3/",
      ],
      "hair_colors": "none",
      "homeworld": "https://swapi.dev/api/planets/31/",
      "language": "Mon Calamarian",
      "name": "Mon Calamari",
      "people": [
        "https://swapi.dev/api/people/27/",
      ],
      "skin_colors": "red, blue, brown, magenta",
      "url": "https://swapi.dev/api/species/8/",
    },
    {
      "average_height": "100",
      "average_lifespan": "unknown",
      "classification": "mammal",
      "created": "2014-12-18T11:22:00.285000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.155000Z",
      "eye_colors": "orange, brown",
      "films": [
        "https://swapi.dev/api/films/3/",
      ],
      "hair_colors": "white, brown, black",
      "homeworld": "https://swapi.dev/api/planets/7/",
      "language": "Ewokese",
      "name": "Ewok",
      "people": [
        "https://swapi.dev/api/people/30/",
      ],
      "skin_colors": "brown",
      "url": "https://swapi.dev/api/species/9/",
    },
    {
      "average_height": "180",
      "average_lifespan": "unknown",
      "classification": "mammal",
      "created": "2014-12-18T11:26:20.103000Z",
      "designation": "sentient",
      "edited": "2014-12-20T21:36:42.157000Z",
      "eye_colors": "black",
      "films": [
        "https://swapi.dev/api/films/3/",
      ],
      "hair_colors": "none",
      "homeworld": "https://swapi.dev/api/planets/33/",
      "language": "Sullutese",
      "name": "Sullustan",
      "people": [
        "https://swapi.dev/api/people/31/",
      ],
      "skin_colors": "pale",
      "url": "https://swapi.dev/api/species/10/",
    },
  ],
}
`;
